state/applicationoffers.go

```diff
@@ -179,11 +195,11 @@ func (s *applicationOffers) removeOps(offerName string) ([]txn.Op, error) {
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
-	zeroRelCount := bson.D{{"relationcount", 0}}
-	zeroOp := txn.Op{
+	relCountAssert := bson.D{{"relationcount", relCount}}
```

Comment:
I don't understand this - the refcount is for relations to the local offered application, right? Why would those need to be unchanged (or zero in the old version) to remove an offer? I can see why you check that there aren't any connections but not why you're checking relation counts.

Reply:
relation count on the app *is* connection count on the offer.
This is the only way know that there are no new connections to the application associated with the offer.

Reply:
Ok, makes sense, thanks.